exporters/filters/multiple_filter.py

```diff
@@ -0,0 +1,116 @@
+from itertools import izip_longest
+from exporters.filters.base_filter import BaseFilter
+from exporters.module_loader import ModuleLoader
+from exporters.utils import dict_list
+
+
+def grouper(n, iterable, fillvalue=None):
+    "grouper(3, 'ABCDEFG', 'x') --> ABC DEF Gxx"
+    args = [iter(iterable)] * n
+    return izip_longest(fillvalue=fillvalue, *args)
+
+
+class MultipleFilter(BaseFilter):
+    """
+    Apply multiple filters to an Item, the filters can be combined. The way
+    this filter works is very similar to [Mongo Query](
+    https://docs.mongodb.com/manual/tutorial/query-documents)
+
+        - filters (list of dict)
+            Filters to be applied to each item
+
+    Example:
+        "filters": [
+            {"name": "exporters.filters.key_value_regex_filter.KeyValueRegexFilter",
+             "options": {
+                 "keys": [
+                     {"name": "country", "value": "United States"}
+                 ]
+             }
+            },
+            {"name": "exporters.filters.key_value_regex_filter.KeyValueRegexFilter",
+             "options": {
+                 "keys": [
+                     {"name": "status", "value": "Verified"}
+                 ]
+             }
+            },
+            {"or": [
+                {"name": "exporters.filters.key_value_regex_filter.KeyValueRegexFilter",
+                 "options": {
+                     "keys": [
+                         {"name": "city", "value": "New York"}
+                     ]
+                 }
+                },
+                {"name": "exporters.filters.key_value_regex_filter.KeyValueRegexFilter",
+                 "options": {
+                     "keys": [
+                         {"name": "city", "value": "Los Angeles"}
+                     ]
+                 }
+                },
+            ]}
+        ]
+
+        The config above could be translated to:
+        country == 'United States' and status == 'Verified' and
+        (city == 'New York' or city == 'Los Angeles')
+    """
+
+    # List of options
+    supported_options = {
+        'filters': {'type': dict_list},
+    }
+
+    def __init__(self, *args, **kwargs):
+        super(MultipleFilter, self).__init__(*args, **kwargs)
+        self.module_loader = ModuleLoader()
+        self.filter_options = self.read_option('filters')
+        self.filters = self._load_filters(self.filter_options)
+        self.logger.info('MultipleFilter instantiated.')
+
+    def filter(self, item):
+        return self._parse_filter(item, self.filters)
```

Comment:
There shouldn't be a need to re-parse the filters for _every_ item from the dataset, that will make a very slow filter.

Parsing the `filters` options should be done only once, when loading the filters.

When loading the multiple filters, this component should:

* parse the filters options
* create a function that will execute the filter

Here is a contrived example of what I'm thinking, using lambdas as filters:

```
def create_or(fs):
    return lambda x: any(f(x) for f in fs)

def combine_filters(filters):
    "Return a function that executes all filters combined"
    def new_filter(x):
        return all(create_or(f['or'])(x) if isinstance(f, dict) else f(x) for f in filters)
    return new_filter
```

This can be used as:

```
>>> even = lambda x: x % 2 == 0
>>> multiple_3 = lambda x: x % 3 != 0
>>> even_or_multiple_3 = combine_filters([{"or": [even, multiple_3]}])
>>> [even_or_multiple_3(x) for x in range(12)]
[True, True, True, False, True, True, True, True, True, False, True, True]
```

So, what I'd expect for MultipleFilters is that `load_filters` would do something like `combine_filters` is doing there, creating a filter function that compiles the filter.
Only, of course, adding proper error handling, using a filter class instead of a callable.
Also, the real implementations should recurse when finding a list, which should also be done before creating the final function.

Does this make sense?